sparkle/src/components/Avatar.tsx

```diff
@@ -144,8 +150,66 @@ export function Avatar({
           {name[0].toUpperCase()}
         </span>
       ) : (
-        <User className="s-h-1/2 s-w-1/2 s-opacity-20" /> // Default icon with size and slate-900 color
+        <User className="s-h-1/2 s-w-1/2 s-opacity-20" />
       )}
     </div>
   );
 }
+
+interface AvatarStackProps {
+  children: ReactNode;
+  maxAvatars?: number;
+}
+
+Avatar.Stack = function ({ children, maxAvatars = 3 }: AvatarStackProps) {
+  const [isHovered, setIsHovered] = useState(false);
+
+  const handleMouseEnter = () => {
+    setIsHovered(true);
+  };
+
+  const handleMouseLeave = () => {
+    setIsHovered(false);
+  };
+
+  // Convert children to an array and slice it to maxAvatars
+  const childrenArray = React.Children.toArray(children);
+  const visibleChildren = childrenArray.slice(0, maxAvatars);
+  const remainingCount = childrenArray.length - maxAvatars - 1;
+
+  return (
+    <div
+      className="s-flex s-flex-row"
+      onMouseEnter={handleMouseEnter}
+      onMouseLeave={handleMouseLeave}
+      style={{
+        marginLeft: "20px",
+      }}
+    >
+      {remainingCount > 0 && (
+        <div
+          className="s-cursor-pointer s-drop-shadow-md"
+          style={{
+            width: isHovered ? "42px" : "32px",
+            transition: "width 300ms ease-out",
+            marginLeft: "-20px",
+          }}
+        >
+          <Avatar size="sm" name={String(remainingCount + 1)} isRounded />
```

Comment:
It took me a minute to figure this out! If I understand correctly the 2 on the left of the second stask here means two more! 
<img width="249" alt="Capture d’écran 2023-09-26 à 17 18 16" src="https://github.com/dust-tt/dust/assets/3803406/67451393-b157-4769-b1fd-588af40fcfee">


Could we try different styles? Maybe the content could be "+2", or maybe it could not look like an avatar? 

Also, following my last comment, if I had 5 Soupinous and I wanted to display only 3, I would call the component as is: 

```javascript

<Avatar.Stack nbMoreItems={2}>
      <Avatar size="sm" name="Soupinou1" visual="ploup" isRounded />
     <Avatar size="sm" name="Soupinou2 visual="ploup" isRounded />
     <Avatar size="sm" name="Soupinou3" visual="ploup" isRounded /> 
</Avatar.Stack>

```

WDYT?

Reply:
> Also, following my last comment, if I had 5 Soupinous and I wanted to display only 3, I would call the component as is:

No strong feeling here.
It makes people recreate the maxNumber logic every time, which feels avoidable.

Reply:
> It makes people recreate the maxNumber logic every time, which feels avoidable.

Yes but less efficient as we are rendering multiple children to then not display them.  The max number logic is not super complex and I much rather handle it manipulating objects than children components. It feels dirty to map among a full array while simultaneously asking the design system via the parent Stack component  to crop the rendering.